linera-core/src/data_types.rs

```diff
@@ -179,6 +179,20 @@ pub struct ChainInfo {
     pub requested_received_log: Vec<ChainAndHeight>,
 }
 
+impl ChainInfo {
+    /// Returns whether this is a child chain that does not have any blocks yet.
+    ///
+    /// Child chains must receive `OpenChain` as their first incoming message.
+    #[allow(clippy::result_large_err)]
+    pub fn needs_opening(&self) -> Result<bool, LocalNodeError> {
+        Ok(self.next_block_height == BlockHeight::ZERO
+            && self
+                .description
+                .ok_or_else(|| LocalNodeError::InactiveChain(self.chain_id))?
```

Comment:
Let's not make this a `LocalNodeError`: This could potentially also be called on a `ChainInfo` obtained from a remote node.

I'd rather keep this inlined, or just return an `Option`, so that the caller can turn it into a `LocalNodeError`.

Reply:
Yes, but at the call site, where we know that this particular `ChainInfo` came from the local node.

If we put this into a method of the _type_ itself, that's an implicit assumption about the call site.